src/sgame/sg_main.cpp

```diff
@@ -2991,9 +2991,20 @@ void G_RunFrame( int levelTime )
 void G_PrepareEntityNetCode() {
```

Comment:
Given that spectating related code is hard to test and your approach is at least as correct as my proposal, I won't insist on putting too much effort into this. Consider adding a comment like this, though: `TODO: Find out whether grouping netcode preparation by spectator state is necessary.`.